docs/rfcs/0003/readme.md

```diff
@@ -0,0 +1,126 @@
+---
+authors: Matt Richerson <matthew.richerson@hpe.com>
+state: discussion
+---
+Storage Profiles for Lustre Layout
+=============================================
+
+Layout of the Lustre targets can have a large effect on file system performance. Currently, the only parameters exposed for tuning a Lustre file system are for the individual target. This RFC proposes adding fields to the NnfStorageProfile to influence where and how many Lustre targets are allocated.
+
+Proposal
+--------
+
+Users may want Lustre file systems with different performance characteristics. For example, a user job with a single compute node accessing the Lustre file system would see acceptable performance from a single OSS. A FPP workload might want as many OSSs as posible to avoid contention. Currently, the WLM does not have enough information in the DirectiveBreakdown to differentiate these two cases.
+
+The NnfStorageProfile will be modified to allow admins to specify where and how many Lustre targets are allocated by the WLM. An end user then chooses an appropriate storage profile for their workload and includes it on the #DW line. During the proposal phase of the workflow, the Rabbit software will use the information in the NnfStorageProfile to add extra constraints to the allocations in the DirectiveBreakdown. The WLM uses these constraints when picking storage.
+
+The NnfStorageProfile is changed to allow three new fields in the mgtOptions, mdtOptions, and ostOptions. The new fields are:
+
+- count - A static value for how many Lustre targets to use.
+- scale - A value from 1-10 that the WLM can use to determine how many Lustre targets to allocate. This is up to the WLM and the admins to agree on how to interpret this field. I imagine a combination of compute node count and allocation size would be used. Using scale=10 as an example, a single compute node job allocating 10TiB might only get 1 OSS. In contrast, a 5000 compute node job allocating 10 TiB might get 20 OSSs. A 5000 compute node job allocating 100TiB might get 100 OSSs.
```

Comment:
Would be nice to have the same cases but with `scale=1`.

Reply:
I can add something. It will just be me making up numbers, though. You guys will have to come up with the inputs and the formula for how to turn this into a meaningful number. It's just an opaque value to us.